File: 브루트포스/N과M(2).py
# 1

N, M = map(int, input().split())


res = [0] * 10
def go(index, start):

    if index == M:
        for i in range(M):
            print(res[i],end=" ")
        print()
        return 

    for i in range(start, N+1):
        res[index] = i
        go(index+1, i+1)

go(0,1)

# 2 

N, M = map(int, input().split())

res = [0] * 10
nums = [x for x in range(1, N+1)]

def go(i, index):
    if index == M:
        for i in range(0, M):
            print(res[i],end=" ")
        print()
        return
    
    if i >= N:
        return

    res[index] = nums[i]
    go(i+1, index+1)

    res[index] = 0
    go(i+1, index)


go(0, 0)

# 3 combinations

from  itertools import combinations

N, M = map(int, input().split())

arr = [x for x in range(1, N+1)]
res = combinations(arr, M)

for nums in res:
  for num in nums:
    print(num, end=" ")
  print()

File: 브루트포스/N과M(2).cpp
#include <iostream>
using namespace std;
// i을 배열에 포함시켰는지 아닌지
// 포함되었으면 true 포함안되었으면 false
bool c[10];
// a배열에 수열들을 담는당
int a[10]; 

void go(int index, int n, int m) {
    if (index == m) {
        for(int i=0 ; i<m ; i++){
            cout << a[i] << " ";
        }
        cout << "\n";
        return;
    }
    
    for(int i=1 ; i<= n ;i++){
        if(c[i]) continue;
        if(index != 0 && a[index-1] > i) continue; 
        a[index] = i;
        c[i] = true; 
        go(index+1, n, m);
        c[i] = false;
    }
}

int main() {
    int n, m;
    cin >> n >> m;
    go(0,n,m);
    return 0;
}